sdks/java/nexmark/src/main/java/org/apache/beam/sdk/nexmark/NexmarkConfiguration.java

```diff
@@ -186,7 +186,14 @@ public void overrideFromOptions(NexmarkOptions options) {
       debug = options.getDebug();
     }
     if (options.getQuery() != null) {
-      query = options.getQuery();
+      try {
+        query = NexmarkQueryName.valueOf(options.getQuery());
+      } catch (IllegalArgumentException exc) {
```

Comment:
If I understand properly, user can specify either `--query=3` or `--query=THREE` and it should end up being the same `NexmarkConfiguration` containing `NexmarkQueryName.THREE`. And you use the exception to differentiate between the two cases. Right ?

Reply:
Yea. The only reason I am keeping the numeric version is to have some backwards compatibility since there's probably lots of dashboards and shell scripts that call this.

Reply:
Yes you were right to keep the numbers also because, as you mentioned, all the dashboards use them and some scripts and docs also.